### ISSUE TITLE
[Feature request] LoRAs sub-folders / sub-menus

### ISSUE THREAD

**Issue by olegsatov** [Acly/krita-ai-diffusion#477]
Hi, 

Would be great to have hierarchy in LoRAs menu. I have more than 50 files there, all in one list. Currently I group them by adding prefixes to filenames, but it's not very convenient. 

The same story - [styles](https://github.com/Acly/krita-ai-diffusion/issues/346). When you have 5-10 styles it's ok to have simple dropdown, but if you often switch between many configurations, this list could be much longer.

**Comment by Acly** [comment]
But Lora menu already is hierachical?

There are few issues with the menu ui, so I don't want to apply it to styles. Would require someone to put in a bit of effort for a better combo box replacement.

**Comment by olegsatov** [comment]
> But Lora menu already is hierachical?

It reads sub-folders, but it shows all the found files in a one long list. Would be great to have hierarchical menu according to folder structure.

> There are few issues with the menu ui, so I don't want to apply it to styles. Would require someone to put in a bit of effort for a better combo box replacement.

I see.

**Comment by Acly** [comment]
Lora should be displayed like seen here: https://github.com/Acly/krita-ai-diffusion/pull/150

**Comment by olegsatov** [comment]
> Lora should be displayed like seen here: #150

This is exactly what I need, but it doesn't work like this in my case. Instead of submenu, I see this:
<img width="376" alt="Снимок экрана 2024-03-10 в 08 01 40" src="https://github.com/Acly/krita-ai-diffusion/assets/147343262/baa8ab4d-3979-4219-a7f6-38c05ccacf30">

Looks like it depends of system. On my Windows machine sub-menus DO WORK. On Mac - they don't. Both machines use the same backend CumfyUI server installed on Windows system by your plugin.

**Comment by Acly** [comment]
I see, it's because Windows server uses backslash `\` and on Mac it's not recognized as directory separator.

**Comment by olegsatov** [comment]
Fix for LoRAs menu on mac. Tested on Mac and Windows. Linux - need to test.

util.php, get_path_dict(), line 143...
`
    new_path_dict = {}
    for path in paths:
        _recurse(new_path_dict, Path(path).parts, path)
        if is_macos or is_linux:
            unix_path = Path(path).as_posix()
            _recurse(new_path_dict, unix_path.split('\\'), path)
        else:
             _recurse(new_path_dict, Path(path).parts, path)
    return new_path_dict
`

<img width="778" alt="Снимок экрана 2024-03-15 в 18 33 15" src="https://github.com/Acly/krita-ai-diffusion/assets/147343262/be276020-3f7c-4511-aa1b-651d5d5bed52">

**Comment by Acly** [comment]
That probably won't work if the server runs on linux. I think we want to "normalize" the incoming path for splitting, maybe like this:
```
    for path in paths:
        parts = Path(path.replace("\\", "/")).parts
        _recurse(new_path_dict, parts, path)
```
This should convert windows server `\` to `/` and leave linux/mac server as is. Should be fine because windows client can handle both.

**Comment by Acly** [comment]
Fixed in [version 1.16.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.16.0)
